jupyter_server_mcp/extension.py

```diff
@@ -75,22 +84,94 @@ def _load_function_from_string(self, tool_spec: str):
             msg = f"Function '{function_name}' not found in module '{module_path}': {e}"
             raise AttributeError(msg) from e
 
-    def _register_configured_tools(self):
-        """Register tools specified in the mcp_tools configuration."""
-        if not self.mcp_tools:
+    def _register_tools(self, tool_specs: list[str], source: str = "configuration"):
+        """Register tools from a list of tool specifications.
+
+        Args:
+            tool_specs: List of tool specifications in 'module:function' format
+            source: Description of where tools came from (for logging)
+        """
+        if not tool_specs:
             return
 
-        logger.info(f"Registering {len(self.mcp_tools)} configured tools")
+        logger.info(f"Registering {len(tool_specs)} tools from {source}")
 
-        for tool_spec in self.mcp_tools:
+        for tool_spec in tool_specs:
             try:
                 function = self._load_function_from_string(tool_spec)
                 self.mcp_server_instance.register_tool(function)
-                logger.info(f"✅ Registered tool: {tool_spec}")
+                logger.info(f"✅ Registered tool from {source}: {tool_spec}")
             except Exception as e:
-                logger.error(f"❌ Failed to register tool '{tool_spec}': {e}")
+                logger.error(f"❌ Failed to register tool '{tool_spec}' from {source}: {e}")
                 continue
 
+    def _discover_entrypoint_tools(self) -> list[str]:
+        """Discover tools from Python entrypoints in the 'jupyter_ai.tools' group.
+
+        Returns:
+            List of tool specifications in 'module:function' format
+        """
+        if not self.use_tool_discovery:
+            return []
+
+        discovered_tools = []
+
+        try:
+            # Use importlib.metadata to discover entrypoints
+            entrypoints = importlib.metadata.entry_points()
+
+            # Handle both Python 3.10+ and 3.9 style entrypoint APIs
+            if hasattr(entrypoints, "select"):
+                tools_group = entrypoints.select(group="jupyter_ai.tools")
```

Comment:
Wondering about the name of the entrypoint group, since `jupyter-server-mcp` can be used without Jupyter AI?